sdks/go/pkg/beam/core/metrics/metrics.go

```diff
@@ -481,7 +481,7 @@ func (m *executionState) kind() kind {
 
 // MsecValue is the value of a single msec metric
 type MsecValue struct {
-	Time int64
+	Time [4]time.Duration
```

Comment:
Consider the following:
We change the type name to DoFnTimes.

Then instead of an array that doesn't indicate any semantics to a user (which has the total? 0? 3?) we expand those to fields.

```
type DoFnTimes struct{
  Start, Process, Finish, Total time.Duration
}
```

This makes it easier for users to understand what they're getting.